src/huggingface_hub/commit_api.py

```diff
@@ -0,0 +1,404 @@
+"""
+Type definitions and utilities for the `create_commit` API
+"""
+
+import base64
+import io
+import os
+import sys
+from concurrent import futures
+from concurrent.futures import ThreadPoolExecutor
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+from typing import BinaryIO, Dict, Iterable, List, Optional, Tuple, Union
+
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+import requests
+
+from .constants import ENDPOINT
+from .lfs import UploadInfo, lfs_upload, post_lfs_batch_info, validate_batch_actions
+from .utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+UploadMode = Literal["lfs", "regular"]
+
+
+@dataclass
+class CommitOperationDelete:
+    """
+    Data strcture holding necessary info to delete
+    a file from a repository on the HF Hub
+
+    Args:
+        path_in_repo (`str`):
+            Path of the file to delete in the repo
```

Comment:
Maybe for consistency with  https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L1750-L1752

```suggestion
        path_in_repo (`str`):
           Relative filepath in the repo, for example:
            `"checkpoints/1fec34a/weights.bin"`
```